scalanet/src/io/iohk/scalanet/peergroup/PeerGroup.scala

```diff
@@ -41,10 +43,22 @@ trait Channel[A, M] {
     */
   def sendMessage(message: M): Task[Unit]
 
+  /**
+    * Consume the next message from the underlying event queue.
+    *
+    * @return The next incoming message, or None if the channel was closed.
+    */
+  def nextMessage(): Task[Option[ChannelEvent[M]]]
+
   /**
     * The inbound stream of messages coming from the remote peer.
+    * Provided for backwards compatibility.
```

Comment:
I wonder do we really need it ? we already have changes interfaces enugh that next update on downstream project will be a lot of work, so forcing us to manaully do `ConnectableSubject.repeatEvalNext(nextMessage())(s)` is not big deal. And this forces this ugly scheduler at interface level.

Reply:
I just didn't want to go and modify all the tests right now, it was good to show that everything still works. We can totally add a new ticket to remove it altogether.